rbi/stdlib/csv.rbi

```diff
@@ -898,6 +930,7 @@ class CSV < Object
   # Also aliased as:
   # [`gets`](https://docs.ruby-lang.org/en/2.7.0/CSV.html#method-i-gets),
   # [`readline`](https://docs.ruby-lang.org/en/2.7.0/CSV.html#method-i-readline)
+  sig { returns(T.nilable(T.any(T::Array[T.untyped], CSV::Row))) }
```

Comment:
We have a fair amount of code at Stripe that assumes the return type from this (and other methods that have similar `T.any(T::Array, CSV::Row)` return types) is a `CSV::Row` and then calls `row["field_name"]`, which now errors with:

```
file.rb:366: Expected T.any(Integer, Float) but found String("query") for argument arg0 http://go/e/7002
     366 |          sql: row["query"],
                             ^^^^^^^
  Expected T.any(Integer, Float) for argument arg0 of method Array#[]:
    https://github.com/sorbet/sorbet/tree/5d0898eb914e813772aceaf0fdc9cb71d2907584/rbi/core/array.rbi#L560:
     560 |        arg0: T.any(Integer, Float),
                  ^^^^
  Got String("query") originating from:
    file.rb:366:
     366 |          sql: row["query"],
                             ^^^^^^^
```

Arguably a lot of these cases should use `T.cast`...but I don't know if that makes this change any more ergonomic or not.